// vertical-stepper-nav/dist/index.js
Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');

/*! *****************************************************************************
Copyright (c) Microsoft Corporation.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
PERFORMANCE OF THIS SOFTWARE.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

var buttonContainerStyles = {
    display: 'inline-flex',
    flexWrap: 'wrap',
    gap: '12px',
    padding: '2px',
    cursor: 'pointer',
};
var Step = function (_a) {
    var stepContent = _a.stepContent, statusColor = _a.statusColor, statusCircleSize = _a.statusCircleSize, onClickHandler = _a.onClickHandler;
    var circleStyles = {
        borderRadius: statusCircleSize !== null && statusCircleSize !== void 0 ? statusCircleSize : 16,
        width: statusCircleSize !== null && statusCircleSize !== void 0 ? statusCircleSize : 16,
        height: statusCircleSize !== null && statusCircleSize !== void 0 ? statusCircleSize : 16,
        border: '2px solid #E1E1E1',
        background: statusColor !== null && statusColor !== void 0 ? statusColor : 'white',
    };
    var keyDownHandler = function (event) {
        if (event.keyCode === 13 || event.keyCode === 32) {
            onClickHandler === null || onClickHandler === void 0 ? void 0 : onClickHandler();
        }
    };
    return (React.createElement("div", { tabIndex: 0, onClick: onClickHandler, onKeyDown: keyDownHandler, role: "button", style: __assign({}, buttonContainerStyles) },
        React.createElement("div", null,
            React.createElement("div", { style: circleStyles })),
        React.createElement("div", { style: { paddingBottom: 2 } }, stepContent())));
};

var separatorStyles = {
    height: '5vh',
    width: 2,
    border: '1px solid #E1E1E1',
    background: '#E1E1E1',
};
var Separator = function (_a) {
    var height = _a.height;
    return React.createElement("div", { style: __assign(__assign({}, separatorStyles), { height: height !== null && height !== void 0 ? height : '5vh' }) });
};

var StepperNav = function (props) {
    return (React.createElement("nav", null, props.steps.map(function (_a, index) {
        var stepContent = _a.stepContent, stepStateColor = _a.stepStateColor, onClickHandler = _a.onClickHandler, stepStatusCircleSize = _a.stepStatusCircleSize;
        return (React.createElement("div", { key: index },
            React.createElement(Step, { stepContent: stepContent, statusColor: stepStateColor, onClickHandler: onClickHandler, statusCircleSize: stepStatusCircleSize }),
            index !== props.steps.length - 1 && (React.createElement("div", { style: {
                    paddingLeft: (stepStatusCircleSize !== null && stepStatusCircleSize !== void 0 ? stepStatusCircleSize : 16) / 2 + 1 + "px",
                } },
                React.createElement(Separator, null)))));
    })));
};

exports.Separator = Separator;
exports.Step = Step;
exports.StepperNav = StepperNav;
